src/services/scope-manager.js

```diff
@@ -0,0 +1,93 @@
+import _ from 'lodash';
+import context from '../context';
+
+// 5 minutes exipration cache
+const SCOPE_CACHE_EXPIRATION_DELTA = 300;
+
+class ScopeManager {
+  constructor({ configStore, forestServerRequester, moment } = context.inject()) {
+    this.configStore = configStore;
+    this.forestServerRequester = forestServerRequester;
+    this.moment = moment;
+    this.scopesCache = {};
+  }
+
+  async getScopeForUser(user, collectionName) {
+    if (!user.renderingId) throw new Error('Missing required renderingId');
+    if (!collectionName) throw new Error('Missing required collectionName');
+
+    const collectionScope = await this._getScopeCollectionScope(user.renderingId, collectionName);
+
+    return ScopeManager._formatDynamicValues(user.id, collectionScope);
+  }
+
+  static _formatDynamicValues(userId, collectionScope) {
+    if (!collectionScope?.scope?.filter) return null;
+
+    // copy scopes in order not to replace the cached values
+    const collectionScopeCopy = _.cloneDeep(collectionScope);
+
+    collectionScopeCopy.scope.filter.conditions.forEach((condition) => {
+      if (condition.value
+        && `${condition.value}`.startsWith('$')
+        && collectionScope.scope.dynamicScopesValues.users[userId]) {
+        condition.value = collectionScope
+          .scope
+          .dynamicScopesValues
+          .users[userId][condition.value];
+      }
+    });
+
+    return collectionScopeCopy.scope.filter;
```

Comment:
You are cloning the big JSON, which contains data about all users just to perform replacements in the filter.
It works, but may be slow.

Did you know about `cloneDeepWith` in lodash?
You can perform substitution without knowing details about the internals of the object.

```suggestion
  return _.cloneDeepWith(collectionScope?.scope?.filter, item => (
    typeof item === 'string' && item.startsWith('$currentUser')
      ? collectionScope.scope.dynamicScopesValues.users[userId][item]
      : undefined
  ));
```